File: app/service/acc/account_master.rb
class Acc::AccountMaster
	def get_data params
		data = Hash.new()
		data["model"] = AccAccountMaster.joins(sdn_admission:[:acd_course_faculty , :acd_degree, :sdn_student]).joins("left join acc_cycle_types on acc_cycle_types.id = acc_account_masters.cycle_type_id")
		
		data["result"] =data["model"].select("acc_account_masters.* , 
			sdn_admissions.admission_no , acc_cycle_types.cycle_name ,
			acd_course_faculties.name course_name, 
			acd_degrees.name degree_name, 
			sdn_students.student_no , 
			sdn_students.date_of_birth,
			concat(sdn_students.first_name,' ',sdn_students.last_name) student_name")

		data
	end
end

File: public/app/model/acc/AccountMaster.js
Ext.define('ExtDesktop.model.acc.AccountMaster', {
    extend: 'Ext.data.Model',
    fields: [
		"id",
		"student_id",
		"admission_id",
		"key_account_id",
		"credit_control_id",
		"fee_cat_id",
		"fee_type",
		"cycle_type_id",
		"account_type",
		"fee_per_cycle",
		"balance",
		"status", 
		"group_account_id",

		"admission_no",
		"cycle_name",
		"course_name",
		"degree_name",
		"date_of_birth",
		"student_no",
		"student_name"
		]
});